diff --git a/ggml/src/ggml-cpu/ggml-cpu-impl.h b/ggml/src/ggml-cpu/ggml-cpu-impl.h
index 0e8dd0a..52806ad 100644
--- a/ggml/src/ggml-cpu/ggml-cpu-impl.h
+++ b/ggml/src/ggml-cpu/ggml-cpu-impl.h
@@ -126,7 +126,7 @@ inline static float vmaxvq_f32(float32x4_t v) {
             MAX(vgetq_lane_f32(v, 2), vgetq_lane_f32(v, 3)));
 }
 
-inline static int32x4_t vcvtnq_s32_f32(float32x4_t v) {
+/*inline static int32x4_t vcvtnq_s32_f32(float32x4_t v) {
     int32x4_t res;
 
     res[0] = roundf(vgetq_lane_f32(v, 0));
@@ -135,7 +135,7 @@ inline static int32x4_t vcvtnq_s32_f32(float32x4_t v) {
     res[3] = roundf(vgetq_lane_f32(v, 3));
 
     return res;
-}
+}*/
 
 inline static uint8x8_t vzip1_u8(uint8x8_t a, uint8x8_t b) {
     uint8x8_t res;
